static/js/util/validation.js

```diff
@@ -269,11 +272,9 @@ export function validateYear(string) {
 
 /**
  * Returns a function which merges the results of the given functions on a set of arguments
- * @param validators {Array} A list of validator functions taking (profile, ui)
- * @returns {function()} A function taking (profile, ui) which combines the outputs of the validators
  */
-export function combineValidators(...validators) {
-  return (...args) => _.merge({}, ...validators.map(
+export function combineValidators(...validators: Function[]): Function {
```

Comment:
I've been playing around with this function trying to make the type checking more specific here. Here's what I have:

    import type { UIState } from '../reducers/ui';
    type ValidatorFunc = (profile: Profile, ui: UIState) => ValidationErrors;
    /**
     * Returns a function which merges the results of the given functions on a set of arguments
     */
    export function combineValidators(...validators: ValidatorFunc[]) : ValidatorFunc {
      return (profile: Profile, ui: UIState) => _.merge({}, ...validators.map(
        validator => validator(profile, ui)
      ));
    }

But I can swap `profile` and `ui` inside `validator` and it doesn't complain. It will complain if I provide a number to the `ui` argument, but if I pass a number to the `profile` argument it doesn't error. :sob: